package/pyproject.toml

```diff
@@ -23,3 +23,90 @@ requires = [
     "setuptools",
     "wheel",
 ]
+
+[project]
+name = "MDAnalysis"
+dynamic = ['version',]
+license = {file = "LICENSE"}
+readmer = "README.rst"
+description = "An object-oriented toolkit to analyze molecular dynamics trajectories."
+authors = [
+    {name = 'MDAnalysis Development Team', email = 'mdanalysis@numfocus.org'}
+]
+maintainers = [
+    {name = 'MDAnalysis Core Developers', email = 'mdanalysis@numfocus.org'}
+]
+requires-python = ">=3.8"
+dependencies = [
+    'numpy>=1.20.0',
+    'biopython>=1.71',
+    'networkx>=2.0',
+    'GridDataFormats>=0.4.0',
+    'mmtf-python>=1.0.0',
+    'joblib>=0.12',
+    'scipy>=1.5.0',
+    'matplotlib>=1.5.1',
+    'tqdm>=4.43.0',
+    'threadpoolctl',
+    'packaging',
+    'fasteners',
+    'gsd>=1.9.3',
+]
+keywords = [
+    "python", "science", "chemistry", "biophysics", "molecular-dynamics",
+    "computational-chemistry", "molecular-simulation", "analysis",
+    "trajectory-analysis",
+]
+classifiers = [
+    'Development Status :: 6 - Mature',
+    'Environment :: Console',
+    'Intended Audience :: Science/Research',
+    'License :: OSI Approved :: GNU General Public License v2 or later (GPLv2+)',
+    'Operating System :: POSIX',
+    'Operating System :: MacOS :: MacOS X',
+    'Operating System :: Microsoft :: Windows',
+    'Programming Language :: Python',
+    'Programming Language :: Python :: 3.8',
+    'Programming Language :: Python :: 3.9',
+    'Programming Language :: Python :: 3.10',
+    'Programming Language :: Python :: 3.11',
+    'Programming Language :: C',
+    'Topic :: Scientific/Engineering',
+    'Topic :: Scientific/Engineering :: Bio-Informatics',
+    'Topic :: Scientific/Engineering :: Chemistry',
+    'Topic :: Software Development :: Libraries :: Python Modules',
+]
+
+[project.optional-dependencies]
+extra_formats = [
+    "netCDF4>=1.0",
+    "h5py>=2.10",
+    "chemfiles>=0.10",
+    "pyedr>=0.7.0",
+]
+analysis = [
+    "seaborn",
+    "scikit-learn",
+    "tidynamics>=1.0.0",
+]
+doc = [
```

Comment:
missing test entry because passing in a dynamic `version` for MDAnalysisTests doesn't really work...